src/Service/SchemaGenerator.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace SilverStripe\MFA\Service;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\MFA\Exception\MemberNotFoundException;
+use SilverStripe\MFA\Extension\MemberExtension;
+use SilverStripe\MFA\Store\StoreInterface;
+use SilverStripe\ORM\FieldType\DBDate;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+use SilverStripe\SiteConfig\SiteConfig;
+
+/**
+ * Generates a multi-factor authentication frontend app schema from the given request
+ */
+class SchemaGenerator
+{
+    use Extensible;
+    use Injectable;
+
+    /**
+     * @var HTTPRequest
+     */
+    protected $request;
+
+    /**
+     * @var StoreInterface
+     */
+    protected $store;
+
+    /**
+     * @param HTTPRequest $request
+     * @param StoreInterface $store
+     */
+    public function __construct(HTTPRequest $request, StoreInterface $store)
+    {
+        $this->setRequest($request);
```

Comment:
Does this need the request as an argument?

Reply:
No, good catch